infinite-tracing/src/main/java/com/newrelic/ResponseObserver.java

```diff
@@ -70,6 +73,14 @@ private boolean isChannelClosing(Throwable t) {
         return t instanceof StatusRuntimeException && t.getCause() instanceof ChannelClosingException;
     }
 
+    /**
+     * Detects if the error received was a connection timeout exception. This can happen if the agent hasn't sent any spans for more than 15 seconds.
+     */
+    protected boolean isConnectionTimeoutException(Throwable t) {
```

Comment:
If this is protected only for the test to have access, let's make it package-private and add a comment or annotation saying that it's available just for testing.